### PR TITLE
add channels and dtype fixtures for testing

### PR DESCRIPTION
**Pull request by johnnv1**
#### Changes

Fixes # (issue)

#### Type of change

- [ ] 📚  Documentation Update
- [x] 🧪 Tests Cases
- [ ] 🐞 Bug fix (non-breaking change which fixes an issue)
- [ ] 🔬 New feature (non-breaking change which adds functionality)
- [ ] 🚨 Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] 📝 This change requires a documentation update

#### Checklist

- [ ] My code follows the style guidelines of this project
- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] I have made corresponding changes to the documentation
- [ ] My changes generate no new warnings
- [ ] Did you update CHANGELOG in case of a major change?

### CODE REVIEW

File: conftest.py [hunk 1]
```diff
@@ -1,129 +0,0 @@
...
-        metafunc.parametrize('dtype_name', dtype_names)
-
-
-def pytest_addoption(parser):
-    parser.addoption('--device', action="store", default="cpu")
-    parser.addoption('--dtype', action="store", default="float32")
-
-
-@pytest.fixture(autouse=True)
-def add_doctest_deps(doctest_namespace):
```
Comment (johnnv1) [inline conftest.py:107]: i forgot, we can not remove this conftest, because of this doctest deps injection

Comment (johnnv1) [reply conftest.py:107]: I'll revert the tests/conftest.py to be within this

File: kornia/color/gray.py [hunk 2]
```diff
@@ -36,9 +36,7 @@ def grayscale_to_rgb(image: Tensor) -> Tensor:
     return concatenate([image, image, image], -3)
 
 
-def grayscale_from_rgb(
-    image: Tensor, rgb_weights: tuple[float, float, float] | None = None, channels_axis: int = -3
-) -> Tensor:
+def grayscale_from_rgb(image: Tensor, rgb_weights: tuple[float, float, float] | None = None) -> Tensor:
```
[the remark this answers is not in the document]
Comment (johnnv1) [reply kornia/color/gray.py:39]: assume that users will use `keras.backend.set_image_data_format('channels_last')` isn't better/cleanest?

[the remark this answers is not in the document]
Comment (edgarriba) [reply kornia/color/gray.py:39]: constraining to the users the channels being -3 or -1 to a fix position every run session could be very limited especially when composing different pipelines coming from different sources

[the remark this answers is not in the document]
Comment (johnnv1) [reply kornia/color/gray.py]: added support for both cases

**Review by edgarriba — APPROVED** [review]

**Pull request merged**